apis/v1/httproute_types.go

```diff
@@ -1362,17 +1362,17 @@ type HTTPCORSFilter struct {
 	// AllowCredentials indicates whether the actual cross-origin request allows
 	// to include credentials.
 	//
-	// The only valid value for the `Access-Control-Allow-Credentials` response
-	// header is true (case-sensitive).
+	// When set to true, the gateway will include the `Access-Control-Allow-Credentials`
+	// response header with value true (case-sensitive).
 	//
-	// If the credentials are not allowed in cross-origin requests, the gateway
-	// will omit the header `Access-Control-Allow-Credentials` entirely rather
-	// than setting its value to false.
+	// When set to false the gateway will omit the header
+	// `Access-Control-Allow-Credentials` entirely (this is the standard CORS
+	// behavior).
 	//
 	// Support: Extended
 	//
 	// +optional
-	AllowCredentials TrueField `json:"allowCredentials,omitempty"`
+	AllowCredentials *bool `json:"allowCredentials,omitempty"`
```

Comment:
Since changing this from bool to an enum of strings would be a breaking change, I agree this is the least bad solution to the problem we have with this field 👍🏻